src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs

```diff
@@ -63,7 +63,7 @@ protected override void SetAbstractValue(AbstractLocation location, DisposeAbstr
             {
                 if (!location.IsNull &&
                     location.LocationTypeOpt != null &&
-                    !location.LocationTypeOpt.IsValueType &&
+                    !location.LocationTypeOpt.HasValueCopySemantics() &&
```

Comment:
I think you can avoid having any value type or value copy semantic check here. `IsDisposable` check just below this line should ensure this.

Reply:
My understand of this check for `IsValueType` was that it has to do with the intent. E.g. if the value is being copied, it should not be tracked regardless of it being disposable or not.

In fact, if this check is completely removed, this would lead a change of behavior in this analyzer for `struct` types, that implement `IDisposable`. From my understanding, because this check is present, they are not being tracked.

Reply:
Sounds reasonable to me.